extensions/ql-vscode/src/variant-analysis/store/repo-states-store.ts

```diff
@@ -1,17 +1,39 @@
 import { outputJson, readJson } from "fs-extra";
 import { VariantAnalysisScannedRepositoryState } from "../shared/variant-analysis";
+import { VariantAnalysisScannedRepositoryStateData } from "./repo-states-data-types";
 
 export const REPO_STATES_FILENAME = "repo_states.json";
 
 export async function writeRepoStates(
   storagePath: string,
-  repoStates: Record<number, VariantAnalysisScannedRepositoryState> | undefined,
+  repoStates: Record<number, VariantAnalysisScannedRepositoryState>,
 ): Promise<void> {
-  return await outputJson(storagePath, repoStates);
+  // Map from repoStates Domain type to the repoStates Data type
+  const repoStatesData = Object.fromEntries(
+    Object.entries(repoStates).map(([key, value]) => {
+      const dataItem: VariantAnalysisScannedRepositoryStateData = value;
+      return [key, dataItem];
+    }),
+  );
+
+  return await outputJson(storagePath, repoStatesData);
 }
 
 export async function readRepoStates(
   storagePath: string,
 ): Promise<Record<number, VariantAnalysisScannedRepositoryState>> {
-  return await readJson(storagePath);
+  const repoStatesData: Record<
```

Comment:
I'm wondering if we need some more error handling here. Now we're no longer just returning the raw data, but doing some processing here.

I think we can assume that if it does exist then it contains valid data, since we're leaving that concern for when we add JSON schema validation later on. But right now what do we do in case that the JSON file doesn't exist?

I've tried to read the documentation for this method and it's not clear to me how it behaves in cases of errors. My best guess by looking at the error handling the existing code has is that it returns `undefined`. It would be good to find some docs, or to test that manually and see.